pkg/network/transports.go

```diff
@@ -39,6 +42,44 @@ func newAutoTransport(v1 http.RoundTripper, v2 http.RoundTripper) http.RoundTrip
 	})
 }
 
+const (
+	initialTO = float64(50 * time.Millisecond)
+	sleepTO   = float64(30 * time.Millisecond)
+	factor    = 1.4
+	numSteps  = 10
+)
+
+var errDialTimeout = errors.New("timed out dialing")
+
+// dialWithBackOff executes `net.Dialer.DialContext()` with exponentially increasing
+// dial timeouts. In addition it sleeps with random jitter between tries.
+func dialWithBackOff(ctx context.Context, network, address string) (net.Conn, error) {
+	return dialBackOffHelper(ctx, network, address, numSteps, initialTO, sleepTO)
+}
+
+func dialBackOffHelper(ctx context.Context, network, address string, steps int, initial, sleep float64) (net.Conn, error) {
```

Comment:
Can you use https://godoc.org/k8s.io/apimachinery/pkg/util/wait#ExponentialBackoff for this? I did something similar in https://github.com/google/go-containerregistry/pull/471/files

Reply:
Nope. It doesn't give me the value of the timeout and that is what I pass to the Dialer.